src/utility/Boards.h

```diff
@@ -729,6 +729,32 @@ void analogWrite(uint8_t channel, uint32_t value, uint32_t valueMax = 255);
 #define PIN_TO_PWM(p)           (p)
 #define PIN_TO_SERVO(p)         (p)
 
+// Raspberry Pi Pico
+// https://datasheets.raspberrypi.org/pico/Pico-R3-A4-Pinout.pdf
+#elif defined(TARGET_RP2040) || defined(TARGET_RASPBERRY_PI_PICO)
+#include <stdarg.h>
+#define TOTAL_ANALOG_PINS       4
+#define TOTAL_PINS              30
+#define VERSION_BLINK_PIN       LED_BUILTIN
+#define IS_PIN_DIGITAL(p)       (((p) >= 0 && (p) < 23) || (p) == LED_BUILTIN)
+#define IS_PIN_ANALOG(p)        ((p) >= 26 && (p) < 26 + TOTAL_ANALOG_PINS)
+#define IS_PIN_PWM(p)           digitalPinHasPWM(p)
+#define IS_PIN_SERVO(p)         (IS_PIN_DIGITAL(p) && (p) != LED_BUILTIN)
+// From the data sheet I2C-0 defaults to GP 4 (SDA) & 5 (SCL) (physical pins 6 & 7)
+// However, v2.3.1 of mbed_rp2040 defines WIRE_HOWMANY to 1 and uses the non-default GPs 6 & 7:
+//#define WIRE_HOWMANY	(1)
+//#define PIN_WIRE_SDA            (6u)
+//#define PIN_WIRE_SCL            (7u)
+#define IS_PIN_I2C(p)           ((p) == PIN_WIRE_SDA || (p) == PIN_WIRE_SCL)
```

Comment:
With this setting, which pins are now used for I2C and SPI? Doesn't this need some special configuration command (I read that the RP2040 has a very flexible pinout)?

Reply:
GPIOs 6 and 7 are used for I2C by the Mbed_rp2040 by default - that's what I tested with. I will check if they can be overridden by redefining in utility/Boards.h.

Reply:
From my investigations, the I2C pins can be set by editing `./variants/RASPBERRY_PI_PICO/pins_arduino.h`. On macOS this is in `~/Library/Arduino15/packages/arduino/hardware/mbed_rp2040/2.3.1`. Yes, the Pico is very flexible regarding which pins are used for I2C. This Boards.h definition assumes on one I2C bus (I2C-1). I can investigate if it is possible to use both I2C buses.

Reply:
I would say the later is optional (although it would of course be nice if one could use more than one I2C bus). 

But we should document that somewhere. Maybe you start a "BoardSupport.md" file in the root directory describing the particulars of the RP2040. Documentation about how to use Firmata with which boards is apparently missing.

Reply:
Added notes on Pico support in a new BoardsSupport.md file.